activerecord/lib/active_record/scoping/named.rb

```diff
@@ -190,7 +190,10 @@ def scope(name, body, &block)
 
         private
           def singleton_method_added(name)
-            generate_relation_method(name) if Kernel.respond_to?(name) && !ActiveRecord::Relation.method_defined?(name)
+            # Most Kernel extends are both singleton and instance methods so
+            # respond_to is a fast check, but we don't want to define methods
+            # only on the module (ex. Module#name)
+            generate_relation_method(name) if Kernel.respond_to?(name) && (Kernel.instance_methods | Kernel.private_instance_methods).include?(name) && !ActiveRecord::Relation.method_defined?(name)
```

Comment:
This generated 3 pretty big arrays on every call:
```ruby
>> (Kernel.instance_methods | Kernel.private_instance_methods).size
=> 119
```

`method_defined?` is preferable:

```suggestion
            generate_relation_method(name) if Kernel.respond_to?(name) && (Kernel.method_defined?(name) || Kernel.private_method_defined?(name)) && !ActiveRecord::Relation.method_defined?(name)
```